template-processors/base/bin/appendResourceVersion.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Kohl's Department Stores, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from pathlib import Path
+import subprocess
+import sys
+import tempfile
+import yaml
+
+# appendResourceVersion - patches the YAML&JSON files in $MANIFEST_DIR,
+# adding the metadata.resourceVersion for each resource being managed.
+# This is intended to serve as a locking mechanism when applying resources
+# in which Kubernetes will fail the apply with a StatusConflict (HTTP status code 409)
+# Ref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency
+#
+# Inputs:
+#
+# MANIFEST_DIR environment variable
+
+if __name__ == "__main__":
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+    logging.info("starting appendResourceVersion")
+    manifest_dir = os.getenv('MANIFEST_DIR')
+    with open('/var/run/secrets/kubernetes.io/serviceaccount/token') as x: token = x.read()
+    files = list(Path(manifest_dir).rglob("*.yml")) + list(Path(manifest_dir).rglob("*.yaml")) + list(Path(manifest_dir).rglob("*.json"))
+    for filename in files:
+        logging.info("processing file {}".format(filename))
+        try:
+            data = yaml.safe_load(subprocess.run(["kubectl",
+                "-s",
+                "https://kubernetes.default.svc:443",
+                "--token",
+                token,
+                "--certificate-authority",
+                "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+                "get",
+                "--ignore-not-found",
+                "-f",
+                filename,
+                "-o",
+                "yaml"], stdout=subprocess.PIPE).stdout)
+            if data is None:
+                logging.error("no kubectl get output for file {}".format(filename))
+                continue
+            if "kind" in data and data["kind"] == "List":
+                logging.info("file {} contains a list".format(filename))
+                if "items" not in data:
```

Comment:
I think `items` would exist in the yaml, but it would be an empty list (example below), so we would want to check that `items` exists and its length is 0.
```
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

```suggestion
                if "items" in data and len(data["items"]) == 0:
```